src/awkward/operations/ak_to_parquet_dataset.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+__all__ = ("to_parquet_dataset",)
+
+
+def to_parquet_dataset(directory, filenames=None, filename_extension=".parquet"):
+    """
+    Args:
+        directory (str or Path): A local directory in which to write `_common_metadata`
+            and `_metadata`, making the directory of Parquet files into a dataset.
+        filenames (None or list of str or Path): If None, the `directory` will be
+            recursively searched for files ending in `filename_extension` and
+            sorted lexicographically. Otherwise, this explicit list of files is
+            taken and row-groups are concatenated in its given order. If any
+            filenames are relative, they are interpreted relative to `directory`.
+        filename_extension (str): Filename extension (including `.`) to use to
+            search for files recursively. Ignored if `filenames` is None.
+
+    Creates a `_common_metadata` and a `_metadata` in a directory of Parquet files.
+
+        >>> ak.to_parquet(array1, "/directory/arr1.parquet", parquet_compliant_nested=True)
+        >>> ak.to_parquet(array2, "/directory/arr2.parquet", parquet_compliant_nested=True)
+        >>> ak.to_parquet_dataset("/directory")
+
+    The `_common_metadata` contains the schema that all files share. (If the files
+    have different schemas, this function raises an exception.)
+
+    The `_metadata` contains row-group metadata used to seek to specific row-groups
+    within the multi-file dataset.
+    """
+
+    # Implementation
+
+    from os import fsdecode, path
```

Comment:
Why restrict to local paths?

Reply:
Addressed above. @zbilodea started with the Awkward 1.x implementation as a model, which was strictly local filesystem. That's something we want to expand here, but I think it can be done in two steps, with this PR being the first step.

Actually, that's Zoë's choice: Zoë, if you want to continue working on this PR to add fsspec now, that's fine.

Reply:
Okay! I think I will continue add fsspec now then